app/controllers/application_controller.rb

```diff
@@ -6,6 +6,15 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   after_action :allow_iframe!
 
+  rescue_from CanCan::AccessDenied do |_exception|
+    if signed_in?
+      redirect_to permission_denied_path
```

Comment:
Easiest fix for this is to redirect to the root_url and set a flash message with a permission denied error